quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -107,6 +111,76 @@ private void doCodeActionsForUnknown(Diagnostic diagnostic, PropertiesModel docu
 		}
 	}
 
+	/**
+	 * Returns the <code>Position</code> to insert the missing required code action property into
+	 * @param textDocument the text document
+	 * @return the <code>Position</code> to insert the missing required code action property into
+	 * @throws BadLocationException
+	 */
+	private Position getPositionForRequiredCodeAction(TextDocument textDocument) throws BadLocationException {
+		String textDocumentText = textDocument.getText();
+		Position position;
+		if (!StringUtils.hasText(textDocumentText)) {
+			position = new Position(0, 0);
+		} else {
+			position = textDocument.positionAt(StringUtils.trimEnd(textDocumentText).length());
+		}
+		return position;
+	}
+
+	/**
+	 * Returns the missing required property name from <code>diagnosticMessage</code>
+	 * @param diagnosticMessage the diagnostic message containing the property name in single quotes
+	 * @return the missing required property name from <code>diagnosticMessage</code>
+	 */
+	private String getPropertyNameFromRequiredMessage(String diagnosticMessage) {
+		String property = diagnosticMessage.substring(diagnosticMessage.indexOf("'") + 1);
+		return property.substring(0, property.indexOf("'"));
+	}
+
+	/**
+	 * Create a code action that inserts all missing required properties and equals signs if there
+	 * are more than one missing required properties.
+	 * 
+	 * @param diagnostics        the diagnostics, one for each missing required property
+	 * @param document           the properties model
+	 * @param formattingSettings the formatting settings
+	 * @param codeActions        the code actions list to fill
+	 */
+	private void doCodeActionForAllRequired(List<Diagnostic> diagnostics, PropertiesModel document,
+			QuarkusFormattingSettings formattingSettings, List<CodeAction> codeActions) {
+		
+		TextDocument textDocument = document.getDocument();
+		List<String> properties = new ArrayList<>();
+		for (Diagnostic diagnostic : diagnostics) {
+			if (ValidationType.required.name().equals(diagnostic.getCode())) {
+				properties.add(getPropertyNameFromRequiredMessage(diagnostic.getMessage()) + (formattingSettings.isSurroundEqualsWithSpaces() ? " = " : "="));
+			}
+		}
+
+		if (properties.isEmpty()) {
+			return;
+		}
+
+		try {
+
+			Position position = getPositionForRequiredCodeAction(textDocument);

```

Comment:
Try 

Position position = document.getDocument().positionAt(document.getEnd());

Reply:
Changing it to 
`Position position = document.getDocument().positionAt(document.getEnd());`

would place the required properties at the end of the file, even if there is trailing whitespace.

For example for this file:
```
quarkus.application.name=
quarkus.application.version=



```

if I do the code action, it will insert it at the end like so:
```
quarkus.application.name=
quarkus.application.version=



quarkus.required.property=
quarkus.required.property=
quarkus.required.property=
```

instead of putting it after the last property like so:
```
quarkus.application.name=
quarkus.application.version=
quarkus.required.property=
quarkus.required.property=
quarkus.required.property=



```

Is that okay?

Reply:
No, it should be after the last property

Reply:
@angelozerr can we keep `getPositionForRequiredCodeAction(TextDocument textDocument)`?

Reply:
Ok I understand, but I would prefer not to use String and trim to retrieve this position. I think it should be better to get the last char of the String and loop for each chars from the last to 0 offset until you find a non whitespace character. After that you convert the offset to a position.